app/llm/external/openai_chat.py

```
@@ -113,20 +171,39 @@ def chat(
-                        messages=messages,
-                        temperature=arguments.temperature,
-                        max_tokens=arguments.max_tokens,
-                        response_format=ResponseFormatJSONObject(type="json_object"),
-                    )
+                    if self.tools:
+                        response = self._client.chat.completions.create(
+                            model=self.model,
+                            messages=messages,
+                            temperature=arguments.temperature,
+                            max_tokens=arguments.max_tokens,
+                            response_format=ResponseFormatJSONObject(type="json_object"),
```

Ruff

Undefined name `ResponseFormatJSONObject`

(F821)